include/rocksdb/utilities/transaction_db.h

```diff
@@ -97,6 +97,12 @@ struct TransactionOptions {
   // Status::Busy.  The user should retry their transaction.
   bool deadlock_detect = false;
 
+  // If set, it states that the CommitTimeWriteBatch represents the latest state
+  // of the application and meant to be used later during recovery. It enables
+  // an optimization to postpone updating the memtable with CommitTimeWriteBatch
+  // to only SwithcMamtable or recovery.
+  bool use_only_the_last_commit_time_batch_for_recovery = false;
```

Comment:
I would prefer we mark CommitTimeWriteBatch() as deprecated and use a separate API to set recoverable state. Something like `Transaction::WriteRecoverableState()`.But it up to you.

Reply:
Sounds like a good idea for long term. If this turns out to be a stable solution we should go with changing the API.

Reply:
sounds good.